### PR TITLE
Add BT permission not granted warning for macOS

### PR DESCRIPTION
**Pull request by beyarkay**
On macOS, the user needs to grant their terminal application permission to access bluetooth, and python will exit without an error message if this permission is not given. Adding a warning here will save confusion in the future for those who just want to get `bleak` up and running.

This is my first time contributing, let me know if there's anything I missed out that I should add!

Pull Request Guidelines for Bleak
---------------------------------

Before you submit a pull request, check that it meets these guidelines:

1. If the pull request adds functionality, the docs should be updated.
2. Modify the `CHANGELOG.rst`, describing your changes as is specified by the
   guidelines in that document.
3. The pull request should work for Python 3.7+ on the following platforms:
    - Windows 10, version 16299 (Fall Creators Update) and greater
    - Linux distributions with BlueZ >= 5.43
    - OS X / macOS >= 10.11
4. Squash all your commits on your PR branch, if the commits are not solving
   different problems and you are committing them in the same PR. In that case,
   consider making several PRs instead.
5. Feel free to add your name as a contributor to the `AUTHORS.rst` file!

### CODE REVIEW

File: docs/scanning.rst [hunk 1]
```diff
@@ -24,6 +24,10 @@ To discover Bluetooth devices that can be connected to:
 
 .. warning:: Do not name your script `bleak.py`! It will cause a circular import error.
 
+.. warning:: On macOS you need to give your terminal permission to access Bluetooth.
```
Comment (dlech) [inline docs/scanning.rst:27]: Some terminal programs are configured correctly for Bluetooth, so this should probably say "you *may* need to ..."

Comment (beyarkay) [reply docs/scanning.rst:27]: Fair enough, will make the edit now

**Comment by beyarkay** [comment]
Okay, have made the change, lemme know about any other issues! happy to help out.

**Comment by dlech** [comment]
This needs to target the `develop` branch rather than the `master` branch.

**Comment by beyarkay** [comment]
Hmm okay I've set the upstream to be `develop`, (see my fork [here](https://github.com/beyarkay/bleak/tree/patch-1) and attached screenshot), but it doesn't seem to be  propagating through to this PR. Sorry if I'm missing something obvious. I can close this PR and open a new one if that would be easiest? 

Otherwise if there's a way to update this PRs upstream then I'm keen to learn how.

Thanks for helping me with this!

Screenshot 2022-07-08 at 22 55 13

**Comment by dlech** [comment]
Apparently you have to click the *Edit* button next to the title of the PR. I went ahead and changed it already.

**Comment by dlech** [comment]
Thanks!

**Comment by beyarkay** [comment]
And thank you! glad to help out.

**Pull request merged**